tracer/test/Datadog.Trace.ClrProfiler.IntegrationTests/YarpDistributedTracingTests.cs

```diff
@@ -0,0 +1,63 @@
+// <copyright file="YarpDistributedTracingTests.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+#if NETCOREAPP3_1_OR_GREATER
+
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Datadog.Trace.Configuration;
+using Datadog.Trace.TestHelpers;
+using FluentAssertions;
+using VerifyXunit;
+using Xunit;
+using Xunit.Abstractions;
+using Xunit.Sdk;
+
+namespace Datadog.Trace.ClrProfiler.IntegrationTests
+{
+    public class YarpDistributedTracingTests : TracingIntegrationTest
+    {
+        private const string ServiceVersion = "1.0.0";
+
+        public YarpDistributedTracingTests(ITestOutputHelper output)
+            : base("Yarp.DistributedTracing", output)
+        {
+            SetServiceVersion(ServiceVersion);
+        }
+
+        public override Result ValidateIntegrationSpan(MockSpan span, string metadataSchemaVersion) => Result.DefaultSuccess;
+
+        [SkippableTheory]
+        [Trait("Category", "EndToEnd")]
+        [Trait("RunOnWindows", "True")]
+        [MemberData(nameof(PackageVersions.Yarp), MemberType = typeof(PackageVersions))]
+        public void SubmitsTraces(string packageVersion)
+        {
+            // We expect the following trace to be generated:
+            // http.request => aspnet_core.request => http.request => aspnet_core.request
+            const int expectedSpans = 4;
+
+            int aspnetCorePort = TcpPortProvider.GetOpenPort();
+
+            using (var agent = EnvironmentHelper.GetMockAgent())
+            using (RunSampleAndWaitForExit(agent, packageVersion: packageVersion, aspNetCorePort: aspnetCorePort))
+            {
+                var spans = agent.WaitForSpans(expectedSpans, 1_000);
+                spans.Count.Should().Be(expectedSpans);
+
+                // All Datadog spans should be in the same trace
+                spans.Select(x => x.TraceId).Distinct().Count().Should().Be(1);
```

Comment:
No snapshots? 🤔

Reply:
Hmm good point. I hadn't thought about it since this automatic instrumentation isn't generating new spans of its own, but that could be helpful to see the exact spans generated. I'll add one in a follow-up commit 👍🏼

Reply:
Added snapshot tests in b59e6ef55c3e420920cc987aaa958666caffcb4b